api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/DetermineEntitlementMessageProcessor.java

```diff
@@ -82,8 +104,16 @@ private void handleDecision(Claim claim, PaymentCycle previousPaymentCycle, Paym
             }
         } else if (claim.getClaimStatus() == ACTIVE) {
             eligibilityDecisionHandler.handleIneligibleDecision(claim, previousPaymentCycle, currentPaymentCycle, decision);
+        } else if (claimHasBeenPendingExpiryForLongerThanTheMaximumDuration(claim, messageTimestamp)) {
+            claim.updateClaimStatus(EXPIRED);
+            claimMessageSender.sendReportClaimMessage(claim, decision.getDateOfBirthOfChildren(), UPDATED_FROM_PENDING_EXPIRY_TO_EXPIRED);
         }
-        //TODO HTBHF-1296: If not ACTIVE, PENDING_EXPIRY will be moved to EXPIRED after 16 weeks.
+    }
+
+    private boolean claimHasBeenPendingExpiryForLongerThanTheMaximumDuration(Claim claim, LocalDateTime messageTimestamp) {
+        // !isAfter has the same effect as (isBefore || isEqual). Unfortunately isBeforeOrEqual does not exist on LocalDateTime.
+        return !claim.getClaimStatusTimestamp().isAfter(messageTimestamp.minus(maximumPendingExpiryDuration))
```

Comment:
I'd suggest swapping the if statement around so that checking for PENDING_EXPIRY comes first.

Reply:
Swapped around.